src/Uno.UI/UI/Xaml/Controls/ToolTip/ToolTip.cs

```diff
@@ -73,10 +129,600 @@ private void AttachToPopup()
+
+				if (bIsRTL)
+				{
+					left = screenWidth - left;
+				}
+
+				// position the parent Popup
+				spPopup.VerticalOffset = top + verticalOffset;
+				spPopup.HorizontalOffset = left + horizontalOffset;
+			}
+		}
+
+		// Sets the location of the ToolTip's Popup within the Xaml window.
+		private void PerformNonMousePlacementWithPopup(
+			Rect? pTargetRect, Rect pDimentions, PlacementMode placement)
+		{
+
+			var tooltipActualWidth = pDimentions.Width;
+			var tooltipActualHeight = pDimentions.Height;
+			var horizontalOffset = pDimentions.Left;
+			var verticalOffset = pDimentions.Top;
+
+			var bIsRTL = false;
+
+			var origin = default(Point);
+			var rcDockTo = default(Rect);
+			var szFlyout = default(Size);
+
+			var spTarget = _owner as FrameworkElement;
+			_Debug.Assert(spTarget != null, "pTarget expected to be non-null in ToolTip_Partial::PerformPlacement()");
+			if (spTarget != null)
+			{
+				// UNO TODO
+				//	        xaml::FlowDirection targetFlowDirection = xaml::FlowDirection_LeftToRight;
+
+				//spTarget->get_FlowDirection(&targetFlowDirection);
+				//bIsRTL = targetFlowDirection == xaml::FlowDirection_RightToLeft;
+
+				// We should not do placement if the target is no longer in the live tree.
+				if (!spTarget.IsLoaded)
+				{
+					return;
+				}
+			}
+
+			var spPopup = _popup;
+			if (spPopup == null)
+			{
+				return;
+			}
+
+			// For ToolTips opened by keyboard focus, PlacementMode_Mouse doesn't make any sense.
+			// Fall back to the default - PlacementMode_Top.
+			if (placement == PlacementMode.Mouse)
+			{
+				placement = PlacementMode.Top;
+			}
+
+			if (spTarget == null)
+			{
+				return;
+			}
+
+			var visibleRect = Windows.UI.Xaml.Window.Current.Bounds;
+			var constraint = visibleRect;
+
+			var windowRect = Windows.UI.Xaml.Window.Current.Bounds;
+			origin.X = windowRect.X;
+			origin.Y = windowRect.Y;
+
+			szFlyout = GetTooltipSize();
+
+			var placementRect = GetPlacementRectInWindowCoordinates();
+
+			var getDockToRectFromTargetElement = false;
+
+			if (pTargetRect.HasValue)
+			{
+				// Slider case - position ToolTip over Thumb rect
+				rcDockTo = pTargetRect.Value;
+			}
+			else if (!placementRect.IsEmpty)
+			{
+				rcDockTo = placementRect;
+			}
+			else if (!m_isSliderThumbToolTip
+				// UNOTODO
+				//&&
+				//      (AutomaticToolTipInputMode::Touch == m_inputMode || AutomaticToolTipInputMode::Mouse == m_inputMode)
+				)
+			{
+				var lastPointerEnteredPoint = default(Point);
+
+				// UNO TODO: PointerPoint.GetCurrentPoint(uint pointerId)
+				lastPointerEnteredPoint = CoreWindow.GetForCurrentThread().PointerPosition;
+
+				rcDockTo.X = lastPointerEnteredPoint.X;
+				rcDockTo.Y = lastPointerEnteredPoint.Y;
+
+				// UNO TODO
+				//// For touch, we need to account for the fact that the context menu hint has a vertical offset.
+				//if (AutomaticToolTipInputMode::Touch == m_inputMode)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5404644)